packages/gestalt/src/Row.js

```diff
@@ -0,0 +1,125 @@
+// @flow
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import FlexBox from './FlexBox.js';
+import Box, {
+  AlignContentPropType,
+  AlignItemsPropType,
+  AlignSelfPropType,
+  DimensionPropType,
+  FlexPropType,
+  JustifyContentPropType,
+  MarginPropType,
+  OverflowPropType,
+  PaddingPropType,
+  type AlignContent,
+  type AlignItems,
+  type AlignSelf,
+  type Dimension,
+  type Flex,
+  type JustifyContent,
+  type Margin,
+  type Overflow,
+  type Padding,
+} from './Box.js';
+
+type Props = {|
+  alignContent?: AlignContent,
+  alignItems?: AlignItems,
+  alignSelf?: AlignSelf,
+  children?: React.Node,
+  fit?: boolean,
+  flex?: Flex,
+  height?: Dimension,
+  justifyContent?: JustifyContent,
+
+  maxHeight?: Dimension,
+  maxWidth?: Dimension,
+  minHeight?: Dimension,
+  minWidth?: Dimension,
+
+  overflow?: Overflow,
+
+  padding?: Padding,
+  smPadding?: Padding,
+  mdPadding?: Padding,
+  lgPadding?: Padding,
+
+  paddingX?: Padding,
+  smPaddingX?: Padding,
+  mdPaddingX?: Padding,
+  lgPaddingX?: Padding,
+
+  paddingY?: Padding,
+  smPaddingY?: Padding,
+  mdPaddingY?: Padding,
+  lgPaddingY?: Padding,
+
+  spacing?: Margin,
+  width?: Dimension,
+  wrap?: boolean,
+|};
+
+export default function Row({
+  alignItems,
+  children,
+  justifyContent,
+  spacing,
+  ...rest
+}: Props) {
+  return (
+    <FlexBox
+      alignItems={alignItems ?? 'center'}
+      direction="row"
+      justifyContent={justifyContent ?? 'start'}
+      {...rest}
+    >
+      {React.Children.map(children, (child, index) => (
+        <Box
+          marginStart={index === 0 ? 0 : spacing}
```

Comment:
@dangerismycat this will cause an issue when `wrap` is set to `true` since the spacing of the individual items will not be correct when it flex wraps:

```
item1  |
  item2  |
  item3|
```

Instead, can we use symmetrical padding?

```js
<Box marginStart={-gap} marginEnd={-gap}>
  {React.Children.map(children, (child, index) => (
    <Box paddingX={gap}>{child}</Box>
  ))}
</Box>
```

Reply:
Yep, good point! I hadn't considering wrapping.